stake-pool/program/src/processor.rs

```diff
@@ -594,6 +617,16 @@ impl Processor {
             msg!("Reserve stake account not in intialized state");
             return Err(StakePoolError::WrongStakeState.into());
         };
+        // Numerator should be smaller than or equal to denominator (fee <= 1)
+        if withdrawal_fee.numerator > withdrawal_fee.denominator {
+            return Err(StakePoolError::FeeTooHigh.into());
+        }
+        // To keep u64 arithmetic sane, we don't allow denominators (and consequently numerators)
+        // that are too large
+        if withdrawal_fee.denominator > 1_000_000 {
+            msg!("Withdrawal fee denominator too large");
+            return Err(StakePoolError::InvalidFeeDenominator.into());
+        }
```

Comment:
What's the reason for this?  If we want to set a max fee denominator of 1 million, we may as well change the whole thing to a `u32`, have it as a `per million` and not use the `Fee` type at all.  For consistency, I'd recommend sticking with the `Fee` type

Reply:
It was to simplify the arithmetic so as not to require checked muls. We need to multiply 3 numbers together to check the fee increase limit, and knowing the bounds means we can be safer in terms of error handling.

Reply:
we're working on a new branch with the `Fee` struct reworked so that we don't need checked arithmetic operations. In brief here's the proposal:

- `numerator` and `denominator` are private `u64` fields
- following restrictions checked on `Fee` initialization: `denominator` >= 1, `numerator` <= `denominator`, `denominator` < some arbitrary precision such that multiplication and division will never overflow `u64`. We're thinking `1_000_000_000`
- impl `PartialOrd` and overload `std::ops::Mul`

so, basically changing `Fee` into a constrained fraction

Reply:
following some discussion with @billythedummy the approach pursued was a bit over the top. 

Decided to just go with u32 fees as I don't think anyone will ever need more than 9 dps of precision.